src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -1110,6 +1109,7 @@ private void ReturnConnectionToPool()
                     sendRequestContentTask.GetAwaiter().GetResult();
                 }
             }
+            Debug.Assert(_writeOffset == 0, "Everything in write buffer should have been flushed.");
```

Comment:
Nit: I think it'd be better if this were moved to be the first line of ReturnConnectionToPoolCore here:
https://github.com/dotnet/corefx/pull/25137/files#diff-f70b71fa62ffc09de0b0d0dcdd33a4bcR1119

Reply:
That is where my first submit put it. But I did actual run the tests with assert in later commit. (line 1112)

The code above calls "innerConnection.ReturnConnectionToPoolCore(); " and I was not sure if that can have any impact. I can re-run the tests easily. Line 1120 makes more sense to me as well.

Reply:
That code calls ReturnConnectionToPoolCore only when the sendRequestContentTask has completed.  If there's still outstanding data to be flushed after it's completed, that's a bug to be fixed, and we'd want the assert to help catch it.

Reply:
ok. I made the modification and started another test run. I'll submit update if it holds till evening.